src/components/learner-credit-management/cards/NewAssignmentModalButton.jsx

```diff
@@ -42,19 +42,34 @@ const NewAssignmentModalButton = ({ enterpriseId, course, children }) => {
   const [assignButtonState, setAssignButtonState] = useState('default');
   const [createAssignmentsErrorReason, setCreateAssignmentsErrorReason] = useState();
   const { displayToastForAssignmentAllocation } = useContext(BudgetDetailPageContext);
+  const { data: subsidyAccessPolicy } = useSubsidyAccessPolicy(subsidyAccessPolicyId);
+  const {
+    subsidyUuid, assignmentConfiguration, isSubsidyActive, isAssignable, catalogUuid,
+  } = subsidyAccessPolicy;
+  const sharedEnterpriseTrackEventMetadata = {
+    subsidyAccessPolicyId,
+    catalogUuid,
+    subsidyUuid,
+    isSubsidyActive,
+    isAssignable,
+    contentPriceCents: course.normalizedMetadata.contentPrice * 100,
+    contentKey: course.key,
+    courseUuid: course.uuid,
+    assignmentConfigurationUuid: assignmentConfiguration.uuid,
+    assignmentConfigurationActive: assignmentConfiguration.active,
+  };
 
   const { mutate } = useAllocateContentAssignments();
-
   const pathToActivityTab = generatePath(routeMatch.path, { budgetId: subsidyAccessPolicyId, activeTabKey: 'activity' });
 
   const handleOpenAssignmentModal = () => {
     open();
     sendEnterpriseTrackEvent(
       enterpriseId,
-      EVENT_NAMES.LEARNER_CREDIT_MANAGEMENT.ASSIGNMENT_MODAL_ASSIGN_COURSE,
+      EVENT_NAMES.LEARNER_CREDIT_MANAGEMENT.ASSIGN_COURSE,
       {
         isOpen: !isOpen,
-        courseUUID: course.uuid,
+        courseUuid: course.uuid,
```

Comment:
[curious] Would it make sense to include `sharedEnterpriseTrackEventMetadata` in this metadata (which contains `courseUuid`?